### PR TITLE
[typescript-fetch] Support error handler in middleware

### PR DESCRIPTION
**Pull request by orange-buffalo**
### Status Quo
`typescript-fetch` provides support for two types of interceptors: to pre-process the request and post-process the response. However, there is one more case which is not handled - network or similar error when response is not received.

### PR suggestion

This PR suggests an additional interceptor hook `onError` that covers the missing piece. It is called in case of exception during request processing and allows to, for instance, re-try the request, replace it with alternative route, etc.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.0.1) (patch release), `6.1.x` (breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by orange-buffalo** [comment]
cc @TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02) @davidgamero (2022/03) @mkusaka (2022/04)

File: modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache [hunk 1]
```diff
@@ -166,7 +166,25 @@ export class BaseAPI {
...
+                        fetch: this.fetchApi,
+                        url: fetchParams.url,
+                        init: fetchParams.init,
+                        error: e,
+                        response: response ? response.clone() : undefined,
+                    }) || response;
+                }
+            }
+            if (!response) {
+                throw new FetchError(e, 'Interceptors did not provide an alternative response');
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache:185]: How about “The request failed and the interceptors did not return an alternative response?

File: modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache [hunk 2]
```diff
@@ -166,7 +166,25 @@ export class BaseAPI {
                 }) || fetchParams;
             }
         }
-        let response = await (this.configuration.fetchApi || fetch)(fetchParams.url, fetchParams.init);
+        let response = null;
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache:169]: Lets use
undefined

File: modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache [hunk 3]
```diff
@@ -166,7 +166,25 @@ export class BaseAPI {
...
+                    response = await middleware.onError({
+                        fetch: this.fetchApi,
+                        url: fetchParams.url,
+                        init: fetchParams.init,
+                        error: e,
+                        response: response ? response.clone() : undefined,
+                    }) || response;
+                }
+            }
+            if (!response) {
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache:184]:
```suggestion
            if (response !== undefined) {
```

**Review by macjohnny — COMMENTED** [review]
Thanks for your contribution

File: modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache (hunk 1 above)
Comment (orange-buffalo) [reply modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache]: Sure, changed

File: modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache (hunk 2 above)
Comment (orange-buffalo) [reply modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache]: No worries, done

File: modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache (hunk 3 above)
Comment (orange-buffalo) [reply modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache]: Applied

**Review by macjohnny — APPROVED** [review]

**Pull request merged**